ONSdigital/ras-rm-secure-message-v2
# Jwt authenication

# What and why?
This add a authentication header to all the endpoints (bar info). It also implements the UAA check, although that needs further work and expanding on another card as requirements are limited on this story.

# How to test?
With UAA up  start this application, now with something like postman call an endpoint which has the decorator on. and provide an Authorization header

![Screenshot 2024-06-24 at 13 50 54](https://github.com/ONSdigital/ras-rm-secure-message-v2/assets/14179817/7d22df61-bfc1-4a88-bd0f-2182acd9b104)

Try the following and see the results

Standard token eyJhbGciOiJIUzI1NiIsInR5cCI6Imp3dCJ9.eyJwYXJ0eV9pZCI6ImNlMTJiOTU4LTJhNWYtNDRmNC1hNmRhLTg2MWU1OTA3MGEzMSIsInJvbGUiOiJpbnRlcm5hbCJ9.ULiRp2GGO94Ax1S4r6vjU3ryBCa_VX6FyXZHIKWSt5E

Missing JWT claims missing role
eyJhbGciOiJIUzI1NiIsInR5cCI6Imp3dCJ9.eyJwYXJ0eV9pZCI6ImNlMTJiOTU4LTJhNWYtNDRmNC1hNmRhLTg2MWU1OTA3MGEzMSJ9.pREGeyWz7xbCBpYNJ8uakDYWQh7sL2MYUy9mIDY2dUQ

JWT claims missing party_id 
eyJhbGciOiJIUzI1NiIsInR5cCI6Imp3dCJ9.eyJyb2xlIjoiaW50ZXJuYWwifQ.iH-5qNb4z_4Y7mB2QrlzHxC3o37DzH4_BSotKKiWXww

The JWT token could not be decoded (wrong key)
eyJhbGciOiJIUzI1NiIsInR5cCI6Imp3dCJ9.eyJwYXJ0eV9pZCI6ImNlMTJiOTU4LTJhNWYtNDRmNC1hNmRhLTg2MWU1OTA3MGEzMSIsInJvbGUiOiJpbnRlcm5hbCJ9.xQ3fzB55Dev3B3oPJsCnre-5-hIsL9HCZOqZRGrvypw

You can also try it with no Authorization header to get "Authorization header is missing" and give it a token that is gibberish that will give a "The JWT token could not be decoded" error
